### PR TITLE
[NETBEANS-4825] - Update Spring license version

### PR DESCRIPTION
**Pull request by pepness**
Update Spring license version, I take into account the [PR-2374](https://github.com/apache/netbeans/pull/2374) from @hectorespert

### CODE REVIEW

**Review by hectorespert — APPROVED** [review]

File: nbbuild/licenses/names.properties [hunk 1]
```diff
@@ -19,9 +19,9 @@ GPL-2-CP=GPL v2 with Classpath exception (http://openjdk.java.net/legal/gplv2+ce
 Apache-2.0=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
 Apache-2.0-groovy=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Apache Groovy addendums
 Apache-2.0-lucene=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Apache Lucene addendums
-Apache-2.0-spring3.2.7=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Spring addendums, version 3.2.7
-Apache-2.0-spring4.0.1=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Spring addendums, version 4.0.1
-Apache-2.0-spring5.2.2=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Spring addendums, version 5.2.2
+Apache-2.0-spring3.2.18=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Spring addendums, version 3.2.18
+Apache-2.0-spring4.3.29=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Spring addendums, version 4.3.29
+Apache-2.0-spring5.2.9=Apache Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt) with Spring addendums, version 5.2.9
```
Comment (matthiasblaesing) [inline nbbuild/licenses/names.properties:24]: I would simplify these to (that needs updates at the use-sites though):

- Apache-2.0-spring3
- Apache-2.0-spring4
- Apache-2.0-spring5

and change the versions at the end to "3.X", "4.X" and "5.X".
-

Comment (pepness) [reply nbbuild/licenses/names.properties]: I agree with your suggestion. Made the changes and commit.

**Comment by matthiasblaesing** [comment]
This fails the license check - the license files need to be renamed too. To verify yourself, you can run:

```shell
ant verify-libs-and-licenses
```

This yields:

```
verify-libs-and-licenses:
[verifylibsandlicenses] /home/matthias/src/netbeans/nbbuild/build/verifylibsandlicenses.xml: 1 failures out of 5 tests
```

```xml
<?xml version="1.0" encoding="UTF-8"?><testsuite errors="0" failures="1" name="org.netbeans.nbbuild.extlibs.VerifyLibsAndLicenses" tests="5" time="0.0">
    <testcase classname="org.netbeans.nbbuild.extlibs.VerifyLibsAndLicenses" name="testNoStrayThirdPartyBinaries" time="0.0"/>
    <testcase classname="org.netbeans.nbbuild.extlibs.VerifyLibsAndLicenses" name="testLicenseFilesAreProperlyFormattedPhysically" time="0.0"/>
    <testcase classname="org.netbeans.nbbuild.extlibs.VerifyLibsAndLicenses" name="testLicenses" time="0.0">
        <failure message="Some license files have incorrect headers" type="junit.framework.AssertionFailedError">Some license files have incorrect headers
java/libs.springframework/external/spring-framework-3.2.18.RELEASE-license.txt refers to nonexistent nbbuild/licenses/Apache-2.0-spring3
java/libs.springframework/external/spring-framework-4.3.29.RELEASE-license.txt refers to nonexistent nbbuild/licenses/Apache-2.0-spring4
java/libs.springframework/external/spring-framework-5.2.9.RELEASE-license.txt refers to nonexistent nbbuild/licenses/Apache-2.0-spring5</failure>
    </testcase>
    <testcase classname="org.netbeans.nbbuild.extlibs.VerifyLibsAndLicenses" name="testBinaryUniqueness" time="0.0"/>
    <testcase classname="org.netbeans.nbbuild.extlibs.VerifyLibsAndLicenses" name="testLicenseinfo" time="0.0"/>
</testsuite>
```

As this has to updated, please rebase onto master and squash changes into one commit.

**Comment by matthiasblaesing** [comment]
Looks sane to me - for the future please follow the conventions for commit messages after squashing. I.e. first line should be a summary, then an empty line and a details description starting in line 3 can follow. For this commit it is ok, but would be good for the future.

**Comment by pepness** [comment]
Will do, thanks for pointing this to me.

File: nbbuild/licenses/names.properties (hunk 1 above)
Comment (red0bear) [reply nbbuild/licenses/names.properties]: can you check too why this .cache/netbeans/12.1/executor-snippets/run.xml:24: is happen when you try compile a swing program ?

**Pull request merged**